### PR TITLE
Favorite stocks

### PR DESCRIPTION
**Pull request by Sakibs**
Implement ability for users to favorite stocks

### CODE REVIEW

File: app/controllers/market_chirp_controller.rb [hunk 1]
```diff
@@ -8,4 +8,5 @@ def index
       @tweets = $twitter.search(@query + ' -rt', result_type: 'mixed', count: 20).take(num)
     end
   end
+
```
Comment (chriskonstad) [inline app/controllers/market_chirp_controller.rb]: Why is this added? Please remove from PR.

File: config/routes.rb [hunk 2]
```diff
@@ -12,6 +12,12 @@
 
   get 'sentiment' => 'sentiment#check'
 
+  get 'favorite' => 'favorite#favorite'
+
+  post 'favorite/create/:company_id' => 'favorite#create_favorite', as: :create_favorite
```
Comment (chriskonstad) [inline config/routes.rb]: I don't think you need the 'as: :create_favorite' for POST paths

File: config/routes.rb [hunk 3]
```diff
@@ -12,6 +12,12 @@
...
+
+  patch 'favorite/remove/:company_id' => 'favorite#remove_favorite', as: :remove_favorite
```
Comment (chriskonstad) [inline config/routes.rb]: See above

**Comment by chriskonstad** [comment]
Also, can you write some tests for it?

File: config/routes.rb (hunk 2 above)
Comment (Sakibs) [inline config/routes.rb]: I use these when calling the path. Its better practice to have rails generate a definition for the path. That way, we only change it in routes.rb and not all the files where we use the path. For example, I use it in the following
          <%= button_to remove_favorite_path(res.id), method: :patch, remote: true, class: "btn btn-default" do%>
This way if we do go back and change the path, we dont need to go and change all instances.

Comment (chriskonstad) [inline config/routes.rb]: Ah okay. Sorry I misunderstood what was going on here, but after reading what you said and checking out the docs, I agree. Good code :)

**Comment by Sakibs** [comment]
Tests added

**Comment by chriskonstad** [comment]
Why do you have copies of commits from master? *Never* merge master into your branch.  If you want to update, run "git checkout master && git pull && git checkout <feature_branch> && git rebase master" to get the latest updates from master.

**Comment by Sakibs** [comment]
I did do git rebase master but it still would run into repeated rebase conflicts and even after rebasing, it would require me to pull. I did research it it suggested I do a merge so I gave it a shot. Ill just port the code over to an new branch then

**Comment by chriskonstad** [comment]
You should never have to pull unless your branch was modified in remote, and never ever ever ever ever ever ever merge no matter what. Just handle the rebase conflicts manually.  The only branch you should ever pull into a feature branch is the remote feature branch

**Comment by Sakibs** [comment]
you mean like https://www.youtube.com/watch?v=Dsc2222N1Iw ?

**Comment by Sakibs** [comment]
Ported over and created new branch
